x/upgrade/abci.go

```diff
@@ -48,18 +48,22 @@ func BeginBlocker(k Keeper, ctx sdk.Context, _ abci.RequestBeginBlock) {
 
```

Comment:
@AdityaSripal we need to signal to IAVL somehow that the previous height (`ctx.BlockHeight() - 1) should be persisted to disk. Is there a way to do that?

/cc @erikgrinaker

Reply:
bump @AdityaSripal, can you envision an IAVL API that would allow us to write a previously committed version to disk?

Reply:
Hmm, so it's already committed (perhaps in memory) and we want a way to tell the disk after the fact to persist to disk?

Yea, that should be fine. Have something like `FlushVersion(version uint64) error`

iAVL checks if it has that version in memory and flushes it to disk. Return an error if the version doesn't exist. e.g. it already got pruned

Reply:
Yes, exactly! Given that we want to do this for the last version (since we panic in `BeginBlock`), we should be guaranteed to have this version in memory. Correct me if I'm wrong @AdityaSripal.

Reply:
Could you guide me in terms of how to implement this API?

Reply:
> Yes, exactly! Given that we want to do this for the last version (since we panic in BeginBlock), we should be guaranteed to have this version in memory.

This should be true. Since keepRecent must be at least 1.

Sure, we basically want to get the root for the specified version, and then do the exact same functionality as `MutableTree.SaveTree` (flush root, flush rootNode branch, flush version orphans),  making sure to flush to disk instead of relying on pruning strategy.
The nodedb.go file has the functions to Save trees to either memory or disk and to get nodes/roots from the db:

```go
// SaveBranch saves the given node and all of its descendants.
// NOTE: This function clears leftNode/rigthNode recursively and
// calls _hash() on the given node.
// TODO refactor, maybe use hashWithCount() but provide a callback.
func (ndb *nodeDB) SaveBranch(node *Node, flushToDisk bool) []byte

func (ndb *nodeDB) saveRoot(hash []byte, version int64, flushToDisk bool) error
```

The implementation of `FlushVersion(version int64)` should do something like the following:

1. Check if version is already in disk , if so return
2. `root = getRoot(version)`, if root = nil, return error
3. `ndb.saveRoot(root, version, true)`
4. `node = GetNode(root)`
5. `ndb.SaveBranch(node, true)`
6. Flush all orphans for that version

Should be able to traverse through the orphans for that version in the memDB (there's a function for this), and flush each one